src/server/middleware/auth.ts

```diff
@@ -523,12 +523,23 @@ const hashRequestBody = (req: FastifyRequest): string => {
  * @returns boolean
  * @async
  */
-const checkIpInAllowlist = async (req: FastifyRequest) => {
+const checkIpInAllowlist = async (
+  req: FastifyRequest,
+): Promise<
+  { isAllowed: false; ip: string } | { isAllowed: true; ip?: string }
```

Comment:
why is `ip` optional in one of the response types?

Reply:
Because I'm not fetching it when there's no allowlist. But honestly it's just reading headers so I'll simplify it.